File: LeetCode_Marathons/LCM2/Easy/Merge_Sorted_Array/a.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        ans_list = []
        id1, id2 = 0, 0
        while id1 + id2 != n + m:
            if id1 == m:
                ans_list.append(nums2[id2])
                id2 += 1
            elif id2 == n or nums1[id1] <= nums2[id2]:
                ans_list.append(nums1[id1])
                id1 += 1
            else:
                ans_list.append(nums2[id2])
                id2 += 1
        
        for i in range(n + m):
            nums1[i] = ans_list[i]

File: LeetCode_Marathons/LCM2/Easy/Merge_Sorted_Array/a.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        ArrayList<Integer> ansMsv = new ArrayList<Integer>();
        int id1 = 0, id2 = 0;

        while (id1 + id2 != n + m) {
            if (id1 == m) {
                ansMsv.add(nums2[id2]);
                id2++;
            } else if (id2 == n || nums1[id1] < nums2[id2]) {
                ansMsv.add(nums1[id1]);
                id1++;
            } else {
                ansMsv.add(nums2[id2]);
                id2++;
            }
        }

        for (int i = 0; i < n + m; i++) nums1[i] = ansMsv.get(i);
    }
}